[profiler] add more CUDA API for kernel launcher

namgyu-youn opened the pull request:
Add more kernel detection options, resolving TODO
- References : [NVIDIA - docs](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__EXECUTION.html)

namgyu-youn committed:
add more CUDA API for kernel launcher

namgyu-youn commented:
@pytorchbot label "release notes: profiler"

namgyu-youn committed:
convert launch_patterns list to tuple

albanD reviewed (approved):
Thanks!

namgyu-youn committed:
fix SyntexError

namgyu-youn commented:
@pytorchbot merge

namgyu-youn commented:
@pytorchmergebot merge -f "lint is green"

namgyu-youn commented:
@pytorchmergebot merge

namgyu-youn commented:
@albanD ; Sorry for the distraction. Could you merge this PR?

albanD commented:
@pytorchbot merge

@namgyu-youn avoid force merging when no test actually ran yet ;)

namgyu-youn committed:
fix AttributeError for CUDA events in profiler
- Since `is_cuda_kernel` checks for `e.name`, it can raise an AttributeError if `e` does not have a `name` attribute.
- For resolving this, we can use `hasattr` to check if `e` has a `name` attribute before accessing it.

namgyu-youn committed:
update utils functions for suggested change

conditioner is not needed

Co-authored-by: albanD <desmaison.alban@gmail.com>

namgyu-youn commented:
@pytorchmergebot merge

namgyu-youn commented:
@pytorchbot merge
